src/full/Agda/Syntax/Translation/ConcreteToAbstract.hs

```diff
@@ -1846,7 +1846,7 @@ instance ToAbstract NiceDeclaration where
           return [ A.DataDef (mkDefInfo x f PublicAccess a r) x' uc (DataDefParams gvars pars) cons ]
       where
         conName (C.Axiom _ _ _ _ _ c _) = return c
-        conName d = errorNotConstrDecl d
```

Comment:
@AlexHarsani This likely introduced regression #6675.
Could you bring back the proper error here (instead of `__IMPOSSIBLE__`)?

Reply:
Ok, I am taking it.

Reply:
No worries, I am on it.

Reply:
Ah sorry, didn't know you took it as you did not reply.
Please have a look at #6678 if that suits you.

Reply:
Oh, sorry, forgot to reply, it's been a busy day. Of course, I'll look into it!